src/operation.rs

```diff
@@ -328,6 +328,21 @@ fn process_unary_operation(
                 ))
             }
         }
+        UnaryOp::FieldsOf() => {
+            if let Term::Record(map) = *t {
+                let mut fields: Vec<String> = map.keys().map(|Ident(id)| id.clone()).collect();
+                fields.sort();
```

Comment:
Is this necessary? Does it mean that our internal representation doesn't have a canonical field order?

Reply:
Our internal representation is Rust's `HashMap`, and the documentation says that `keys()` visits the keys in arbitrary order. Nix does the same thing: `attrNames` returns a sorted list, probably for the same reason (grep for `attrNames` [here](https://nixos.org/manual/nix/stable/)).

Reply:
Ok, this is quite fair.